repos/system_upgrade/common/actors/checktargetiso/libraries/check_target_iso.py

```diff
@@ -0,0 +1,182 @@
+import os
+
+from leapp import reporting
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.common.config import version
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import StorageInfo, TargetOSInstallationImage
+
+
+def inhibit_if_not_valid_iso_file(iso):
+    inhibit_title = None
+    target_os = 'RHEL {}'.format(version.get_target_major_version())
+    if not os.path.exists(iso.path):
+        inhibit_title = 'Provided {target_os} installation ISO does not exists.'.format(target_os=target_os)
+        inhibit_summary_tpl = 'The supplied {target_os} ISO path \'{iso_path}\' does not point to an existing file.'
+        inhibit_summary = inhibit_summary_tpl.format(target_os=target_os, iso_path=iso.path)
+    else:
+        try:
+            # TODO(mhecko): Figure out whether we will keep this since the scanner actor is mounting the ISO anyway
+            file_cmd_output = run(['file', '--mime', iso.path])
+            if 'application/x-iso9660-image' not in file_cmd_output['stdout']:
+                inhibit_title = 'Provided {target_os} installation image is not a valid ISO.'.format(
+                        target_os=target_os)
+                summary_tpl = ('The provided {target_os} installation image path \'{iso_path}\''
+                               'does not point to a valid ISO image.')
+                inhibit_summary = summary_tpl.format(target_os=target_os, iso_path=iso.path)
+
+        except CalledProcessError as err:
+            raise StopActorExecutionError(message='Failed to check whether {0} is an ISO file.'.format(iso.path),
+                                          details={'details': '{}'.format(err)})
+    if inhibit_title:
+        remediation_hint = ('Check whether the supplied target OS installation path points to a valid'
+                            '{target_os} ISO image.'.format(target_os=target_os))
+
+        reporting.create_report([
+            reporting.Title(inhibit_title),
+            reporting.Summary(inhibit_summary),
+            reporting.Remediation(hint=remediation_hint),
+            reporting.Severity(reporting.Severity.MEDIUM),
+            reporting.Groups([reporting.Groups.INHIBITOR]),
+            reporting.Groups([reporting.Groups.REPOSITORY]),
+        ])
+        return True
+    return False
+
+
+def inhibit_if_failed_to_mount_iso(iso):
+    if not iso.was_mounted_successfully:
```

Comment:
Nit: It seems more natural to have the check visa versa 
```
if iso.was_mounted_successfully:
    return False
...
return True
```

Reply:
Resolved by accepting the suggested modification.